sql/src/test/java/io/crate/execution/engine/window/WindowDefinitionTest.java

```diff
@@ -58,6 +58,66 @@ public void prepare() throws IOException {
             .build();
     }
 
+    @Test
+    public void test_offset_preceding_does_not_support_interval_in_rows_mode() {
+        expectedException.expectMessage(containsString("In ROWS mode the offset must be a non-null, non-negative number"));
+        e.analyze(
+            "select id, avg(date::long) over(twenty_sec_prior)" +
+            "  from doc.users" +
+            "  window twenty_sec_prior as (rows between '20 seconds'::interval preceding and current row)" +
+            "  order by id");
+    }
+
+    @Test
+    public void test_offset_preceding_does_not_support_negative_values_in_rows_mode() {
+        expectedException.expectMessage(containsString("In ROWS mode the offset must be a non-null, non-negative number"));
+        e.analyze(
+            "select id, avg(date::long) over(twenty_sec_prior)" +
+            "  from doc.users" +
+            "  window twenty_sec_prior as (rows between -2000 preceding and current row)" +
+            "  order by id");
+    }
+
+    @Test
+    public void test_offset_following_does_not_support_interval_in_rows_mode() {
+        expectedException.expectMessage(containsString("In ROWS mode the offset must be a non-null, non-negative number"));
+        e.analyze(
+            "select id, avg(date::long) over(twenty_sec_following)" +
+            "  from doc.users" +
+            "  window twenty_sec_following as (rows between current row and '20 seconds'::interval following)" +
+            "  order by id");
+    }
+
+    @Test
+    public void test_offset_following_does_not_support_negative_values_in_rows_mode() {
+        expectedException.expectMessage(containsString("In ROWS mode the offset must be a non-null, non-negative number"));
+        e.analyze(
+            "select id, avg(date::long) over(twenty_sec_following)" +
+            "  from doc.users" +
+            "  window twenty_sec_following as (rows between current row and -2000 following)" +
+            "  order by id");
+    }
+
+    @Test
+    public void test_offset_preceding_supports_interval_in_range_mode() {
+        e.analyze(
+            "select id, avg(ints) over(thirty_secs_preceding)" +
+            "  from doc.users" +
+            "  window thirty_secs_preceding as (order by date range between '30 seconds'::interval preceding and current row)" +
+            "  order by id");
+    }
+
+    @Test
+    public void test_offset_following_supports_interval_in_range_mode() {
+        e.analyze(
+            "select id," +
+            "       avg(ints) over(thirty_secs_following)," +
+            "       avg(date::long) over(thirty_secs_following) as \"avg date 30 seconds following (range mode)\"" +
+            "  from doc.users" +
+            "  window thirty_secs_following as (order by date range between current row and '30 seconds'::interval following)" +
+            "  order by id");
+    }
```

Comment:
These tests don't do much except making sure that the query can be analysed somehow.

When I asked to transform the integration tests to unit tests I meant to utilize `AbstractWindowFunctionTest` so that the function execution / windowing logic is executed and tested.